Maple2.Server.Game/Manager/ShopManager.cs

```diff
@@ -0,0 +1,609 @@
+        if (!shop.DisableBuyback) {
+            session.Send(ShopPacket.BuyBackItemCount((short) buyBackItems.Count));
+            if (buyBackItems.Count > 0) {
+                session.Send(ShopPacket.LoadBuyBackItem((BuyBackItem[]) buyBackItems.Values));
+            }
+        }
+    }
+
+    private Shop GetInstancedShop(Shop serverShop) {
+        if (instancedShops.TryGetValue(serverShop.Id, out Shop? instancedShop)) {
+            if (instancedShop.RestockTime < DateTime.Now.ToEpochSeconds()) {
+            }
+            return instancedShop;
+        }
+
+
+        if (!TryGetShopData(serverShop.Id, out CharacterShopData? data)) {
+            long restockTime = GetRestockTime(serverShop.RestockData!.Interval);
+            if (serverShop.RestockTime > 0) {
+                restockTime = serverShop.RestockTime;
+            }
+            return CreateInstancedShop(serverShop, CreateShopData(serverShop), restockTime);
+
+        }
+        if (data.RestockTime < DateTime.Now.ToEpochSeconds()) {
+            return CreateInstancedShop(serverShop, CreateShopData(serverShop), GetRestockTime(serverShop.RestockData!.Interval));
+        }
+
+        // Assemble shop
+        instancedShop = serverShop.Clone()!;
+        instancedShop.RestockTime = data.RestockTime;
+
+        if (!TryGetShopItemData(instancedShop.Id, out IDictionary<int, CharacterShopItemData>? dataDictionary)) {
+            dataDictionary = CreateShopItemData(serverShop);
+        }
+
+        foreach ((int shopItemId, CharacterShopItemData itemData) in dataDictionary) {
+            if (!serverShop.Items.TryGetValue(shopItemId, out ShopItem? shopItem)) {
+                continue;
+            }
+
+            ShopItem itemCopy = shopItem.Clone();
+            itemCopy.Item = itemData.Item;
+            itemCopy.StockPurchased = itemData.StockPurchased;
+            instancedShop.Items.Add(shopItemId, itemCopy);
+        }
+
+        instancedShops[instancedShop.Id] = instancedShop;
+        return instancedShop;
+    }
+
+    private Shop CreateInstancedShop(Shop shop, CharacterShopData shopData, long restockTime) {
+        Shop instancedShop = shop.Clone()!;
+
+        // First delete any existing shop item data
+        long ownerId = shop.RestockData!.PersistantInventory ? session.AccountId : session.CharacterId;
+        using GameStorage.Request db = session.GameStorage.Context();
+
+        if (TryGetShopItemData(shop.Id, out IDictionary<int, CharacterShopItemData>? shopItemDatas)) {
+            foreach ((int shopItemId, CharacterShopItemData itemData) in shopItemDatas) {
+                db.DeleteCharacterShopItemData(ownerId, shopItemId);
+            }
+        }
+
+        // Create new shop data
+        instancedShop.Items = GetShopItems(shop);
+        instancedShop.RestockTime = restockTime;
+        instancedShop.RestockData!.RestockCount = shopData.RestockCount;
+        CreateShopItemData(instancedShop);
```

Comment:
return value of this function is unused?

Reply:
This function is used in two instances: Creating an instanced shop and getting an instanced shop. We're only using this function to create the shop item data entries and not assigning upon creation because it would be an exact identical copy of the current shop.